galleon-plugins/src/main/java/org/wildfly/galleon/plugin/EE9ArtifactInstaller.java

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright 2016-2021 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.wildfly.galleon.plugin;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.StandardCopyOption;
+import java.util.Set;
+import org.jboss.galleon.ProvisioningException;
+import org.jboss.galleon.runtime.ProvisioningRuntime;
+import org.jboss.galleon.universe.maven.MavenArtifact;
+import org.jboss.galleon.universe.maven.MavenUniverseException;
+import org.wildfly.galleon.plugin.WfInstallPlugin.ArtifactResolver;
+import org.wildfly.galleon.plugin.transformer.JakartaTransformer;
+
+/**
+ * Installer that copies already transformed artifact from the
+ * provisioningRepository and compute transformed version for thin server.
+ * Such installer is used for fat and thin server with transformation disabled
+ * and provisioning maven repository being set.
```

Comment:
I suggest adding a sentence here about the expected use case(s) where this would be the situation. The distinction between this and EE9ArtifactTransformerInstaller is not obvious (particularly since this might do some transformation in the override case) so explaining the use case might help.

AIUI this is about the case where the transformed repo has already been created and the goal is to reuse it, e.g. on the WF testsuite.

Calling this "ExistingEE9ArtifactInstaller" might help, but otoh that's not correct for this override transformation case. So I'm not proposing that.